realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -441,33 +441,45 @@ private void emitValidateTableMethod(JavaWriter writer) throws IOException {
 
                 // make sure that nullability matches
                 if (metadata.isNullable(field)) {
-                    writer.beginControlFlow("if (!table.isColumnNullable(%s))", fieldIndexVariableReference(field));
-                    if (Utils.isBoxedType(fieldTypeCanonicalName)) {
+                    // Check if the existing PrimaryKey does support null value for String, Byte, Short, Integer, & Long
+                    if (field.equals(metadata.getPrimaryKey()) && Utils.isNullablePrimaryKeyType(field)) {
```

Comment:
I was mostly refering to `Utils.isNullablePrimaryKeyType(field)`.  I agree that having a seperate error message would be nice, but `if (field.equals(metadata.getPrimaryKey()) && ClassMetaData.isNullable(field))` should to the same? Note I cannot remember how to check for general nullability on the top of my head. But hopefully you get my point :)